src/pages/Profile.tsx

```diff
@@ -159,18 +170,37 @@ const Profile = () => {
             <Box sx={styles.profileContent}>
               <Box sx={styles.avatarContainer}>
                 <Avatar sx={styles.avatar} />
-                {profileContents.username === user.username ? (
-                  <Button
-                    size="small"
-                    variant="outlined"
-                    startIcon={<EditIcon />}
-                    sx={styles.editProfileButton}
-                  >
-                    Edit Profile
-                  </Button>
-                ) : (
-                  <FollowButton />
-                )}
+                {!loading &&
+                  (profileContents.username === user.username ? (
```

Comment:
If we do swap to using user ids change that here too

Reply:
I needed to provide an initial number because it is a part of the profileContents object, and I decided to use `NaN` instead of making it undefined or null because that changes the type of userId from `number` to `number | undefined` or something.

I wanted to only render when userID is a non-zero number so this is the change:

```suggestion
              {!loading && profileContents.userId &&
                  (profileContents.userId === user.userId ? (
```

Reply:
Discussed this offline, we'll go with using `undefined` since that is a better descriptor for the situation of there not being any userId and holds up better under number checks